src/groups/semidirect_product_group.jl

```diff
@@ -64,24 +64,27 @@ function show(io::IO, G::SemidirectProductGroup)
     )
 end
 
+_subarray(::SemidirectProductGroup, x, ::Val{i}) where {i} = submanifold_component(x, i)
+_subarray(G::SemidirectProductGroup, x, i::Int) = _subarray(G, x, Val(i))
```

Comment:
Instead of this function I think it's a better idea to directly call `submanifold_component`. Probably with new methods that support `Val(i)` indexing.

BTW, I don't think the second method of `_subarray` helps under any circumstances. Either everything is inlined and constant-propagated or not and wrapping an argument in `Val` doesn't improve performance.

Reply:
The issue with `submanifold_component` is that we can't dispatch on the manifold. Given the name, perhaps that's something we should change.

It would also be good to have something like `submanifold_components` that dispatches on a manifold and for `ProductArray` and `ProductRepr` returns the `parts` field. For the SE(N) case with matrix input, it would return views of the two subarrays. For a group manifold with `Identity` input, it would return a tuple with `Identity` of the subgroups.

These custom methods are necessary either way though, because there's structure outside the subarrays in the matrix that needs to be enforced for the matrix to still be affine.

Reply:
> The issue with `submanifold_component` is that we can't dispatch on the manifold. Given the name, perhaps that's something we should change.

I think it's better to add a new argument to `submanifold_component` then and simplify `_subarray`.

> It would also be good to have something like `submanifold_components` that dispatches on a manifold and for `ProductArray` and `ProductRepr` returns the `parts` field. For the SE(N) case with matrix input, it would return views of the two subarrays. For a group manifold with `Identity` input, it would return a tuple with `Identity` of the subgroups.

OK, we can have that.